src/index/CompressedRelation.cpp

```diff
@@ -663,6 +663,11 @@ DecompressedBlock CompressedRelationReader::readPossiblyIncompleteBlock(
       return DecompressedBlock{config.scanColumns_.size(), allocator_};
     }
   }();
+  if (manuallyDeleteGraphColumn) {
+    auto tie = [](const auto& row) { return std::tie(row[0], row[1], row[2]); };
+    auto unique = ::ranges::unique(block, ::ranges::equal_to{}, tie);
+    block.erase(unique, block.end());
+  }
```

Comment:
If possible I'd prefer a solution that doesn't initialize `specForAllColumns` with `std::nullopt` everywhere, but that is easier said than done I fear.